src/lavinmq/vhost.cr

```diff
@@ -112,6 +112,9 @@ module LavinMQ
       headers = msg.properties.headers
       find_all_queues(ex, msg.routing_key, headers, visited, found_queues)
       headers.delete("BCC") if headers
+
+      @replicator.wait_for_max_lag
```

Comment:
Is it here we should wait? Not anywhere in `Replication::Server`?

Reply:
worth discussing.. have not thought about having it wait somewhere else, what are you thinking?

Reply:
Dunno :)
Maybe in `action_loop`? I just feel that replication should be responsible for it, not `VHost`. But since replication code is new territory for me, I have no clue really.

Reply:
`action_loop` belongs to the follower, and we want to block publishes on the entire server depending on the perspective of the leader server.. maybe we could put it somewhere in the replicator, but i think it is also pretty clear to put it in the vhosts `publish` method.. but maybe unclear since we do not block publishes per vhost, but on the whole server

Reply:
We now block in the replication server instead: Unfortunately we can't place the block directly in the each_follower function, because the replace function is run when launching, and we will have had no chance to connect any followers resulting in deadlock. Therefore i have placed the block (`wait_for_max_lag`) it in the `append` and `delete_file` functions.